.github/workflows/prepare-release.yml

```diff
@@ -0,0 +1,178 @@
+name: Prepare for a release
+
+on:
+  workflow_dispatch:
+    inputs:
+      component:
+        type: choice
+        options:
+        - OpenTelemetry.Exporter.Geneva
+        - OpenTelemetry.Exporter.InfluxDB
+        - OpenTelemetry.Exporter.Instana
+        - OpenTelemetry.Exporter.OneCollector
+        - OpenTelemetry.Exporter.Stackdriver
+        - OpenTelemetry.Extensions
+        - OpenTelemetry.Extensions.AWS
+        - OpenTelemetry.Extensions.Enrichment
+        - OpenTelemetry.Instrumentation.AspNet
+        - OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModule
```

Comment:
This is kid of misleading. This packages are released always together.

Similar thing for `OpenTelemetry.PersistentStorage.*`

Reply:
Agree. Really this list should be the tag prefixes not components. I decided to go with components though because a) in most cases that is what users will be looking for (probably?) and b) it makes it easy to keep this list in sync with the other places it is mirrored (issue templates). I was hoping there was a way in yaml to pull in a list of items from another file so we could have a single source of truth for the set of known components but it doesn't seem possible 😢

How it works is you pick a component. The workflow will look for a csproj for that component. Then it will find the `MinVerTagPrefix` for that project. The release is then performed for that `MinVerTagPrefix`. So when it comes to `OpenTelemetry.Instrumentation.AspNet*` and `OpenTelemetry.PersistentStorage*` you have to kick off the job for one of the components in the set (doesn't matter which) and the job will release both. It is a bit odd, but I think the benefits outweigh the quirk(s)?

Let me respond to some of the other comments on here...

> Based on what I see, the whole process has to be either fully handled by new pipeline or fully in the old way.

Not true! When you kick off the job a PR will be opened to update CHANGELOGs and public api files (for stable releases). Someone has to merge that. Then the bot will ask you if you want it to create the tag. If you choose to not have the bot create the tag, you can do everything the manual way. Probably easier to let the bot do it, but it isn't forced 😄

> Allow created end-users requests as described in

Agree with this. To kick off this new job you need to be a collaborator in the repo (write access). This will help maintainers but it won't be widely useful. What I am thinking is add an issue template for "Request release" end users may use to request a release be done. I'm hoping the flow will be users select a component, and then the bot can look for code owners for the selected component to approve before kicking off the job. That would give us a nice flow/mechanism. Going to look at that as a follow-up.

> Bulk releases

I think this is already covered by: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/actions/workflows/core-version-update.yml

Maintainers can kick that workflow off anytime a core release happens and it will generate the PR to bump all the components. My goal is to have the main repo automatically call this but that isn't in place yet.